clarity/evaluator/haspi/ebm.py

```diff
@@ -1,302 +1,357 @@
 
     Returns:
-        tuple: xLP - LP filtered and subsampled reference signal envelope
-           Each frequency band is a separate column. yLP - LP filtered and subsampled
-           degraded signal envelope
+    tuple: reference_env - LP filtered and subsampled reference signal envelope
+           Each frequency band is a separate column.
+           processed_env - LP filtered and subsampled degraded signal envelope
 
 
+    Updates:
     James M. Kates, 12 September 2019.
     Translated from MATLAB to Python by Zuzanna Podwinska, March 2022.
     """
     # Check the filter design parameters
-    assert fsub <= fsamp, "Error in ebm.EnvFilt: Subsampling rate too high."
-    assert fcut <= 0.5 * fsub, "Error in ebm.EnvFilt: LP cutoff frequency too high."
+    assert (
+        freq_sub_sample <= freq_samp
+    ), "Error in ebm.EnvFilt: Subsampling rate too high."
+    assert (
+        filter_cutoff <= 0.5 * freq_sub_sample
+    ), "Error in ebm.EnvFilt: LP cutoff frequency too high."
 
     # Check the data matrix orientation
     # Require each frequency band to be a separate column
-    nrow = xdB.shape[0]  # number of rows
-    ncol = xdB.shape[1]  # number of columnts
+    nrow = reference_db.shape[0]  # number of rows
+    ncol = reference_db.shape[1]  # number of columnts
     if ncol > nrow:
-        xdB = xdB.T
-        ydB = ydB.T
-    nsamp = xdB.shape[0]
+        reference_db = reference_db.T
+        processed_db = processed_db.T
+    nsamp = reference_db.shape[0]
 
     # Compute the lowpass filter length in samples to give -3 dB at fcut Hz
-    tfilt = 1000 * (1 / fcut)  # filter length in ms
+    tfilt = 1000 * (1 / filter_cutoff)  # filter length in ms
     tfilt = 0.7 * tfilt  # Empirical adjustment to the filter length
-    nfilt = round(0.001 * tfilt * fsamp)  # Filter length in samples
+    nfilt = round(0.001 * tfilt * freq_samp)  # Filter length in samples
     nhalf = floor(nfilt / 2)
     nfilt = int(2 * nhalf)  # Force an even filter length
 
     # Design the FIR LP filter using a von Hann window to ensure that there are no negative
     # envelope values. The MATLAB code uses the hanning() function, which returns the Hann
     # window without the first and last zero-weighted window samples,
     # unlike np.hann and scipy.signal.windows.hann; the code below replicates this behaviour
-    w = 0.5 * (1 - np.cos(2 * np.pi * np.arange(1, nfilt / 2 + 1) / (nfilt + 1)))
-    benv = np.concatenate((w, np.flip(w)))
+    window = 0.5 * (1 - np.cos(2 * np.pi * np.arange(1, nfilt / 2 + 1) / (nfilt + 1)))
+    benv = np.concatenate((window, np.flip(window)))
     benv = benv / np.sum(benv)
 
     # LP filter for the envelopes at fsamp
-    xenv = convolve2d(xdB, np.expand_dims(benv, 1), "full")  # 2D convolution
-    xenv = xenv[nhalf : nhalf + nsamp, :]
-    yenv = convolve2d(ydB, np.expand_dims(benv, 1), "full")
-    yenv = yenv[nhalf : nhalf + nsamp, :]
+    reference_env = convolve2d(
```

Comment:
Remove the '2D convolution' comment. Perhaps this was less obvious in the original MATLAB code :-)